pkg/lib/credentials.js

```diff
@@ -197,8 +201,13 @@ function Keys() {
             proc.close("terminated");
         }, 10 * 1000);
 
-        proc = cockpit.spawn(["ssh-keygen", "-p", "-f", name],
-                             { pty: true, environ: ["LC_ALL=C"], err: "out", directory: self.path })
+        var cmd = ["ssh-keygen", "-f", file];
+        if (new_type)
+            cmd.push("-t", new_type);
+        else
+            cmd.push("-p");
```

Comment:
This code reads a little weird now -- if we specify a type, then `old_pass` Should Never Happen™, right? This could be a `console.assert()` for robustness, but see below.